hail/src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -221,7 +221,7 @@ object MatrixTable {
     new MatrixTable(kt.hc, matrixType,
       kt.globals,
       BroadcastIndexedSeq(Array.empty[Annotation], TArray(matrixType.colType), kt.hc.sc),
-      OrderedRVD.coerce(matrixType.orvdType, rdd))
+      rvd.changeKey(matrixType.rowKey))
```

Comment:
Oh, you're right, this was a completely unnecessary coerce. I guess it was there from when tables weren't ordered, and I didn't notice it could be removed now.